applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -245,6 +226,25 @@ def InitializeSolutionStep(self):
     def FinalizeSolutionStep(self):
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Mass and momentum conservation equations are solved.")
 
+        # Recompute the distance field according to the new level-set position
+        if (self._reinitialization_type == "variational"):
+            self._GetDistanceReinitializationProcess().Execute()
+        elif (self._reinitialization_type == "parallel"):
+            adjusting_parameter = 0.05
+            layers = int(adjusting_parameter*self.main_model_part.NumberOfElements()) # this parameter is essential
```

Comment:
Just wondering, why so many layers? I think that in the past we used to use something around 10.

Reply:
I have normalized the number of layers with the number of elements. Unfortunately, this re-distancing scheme needs too many iterations in order to get a desirable result.

Reply:
Actually we cannot ensure a priori that the resultant distance is real distance "enough" for the next convection right?

Reply:
Yes. That is the point. As far as I checked, if the level-set function is not (let's say) regular enough over the whole domain, the result would be affected.